cli/awsbatch/awsbsub.py

```diff
@@ -164,82 +163,62 @@ def _validate_parameters(args):
-        :param s3_bucket: S3 bucket to use
-        """
-        self.boto3_factory = boto3_factory
-        self.s3_bucket = s3_bucket
-
-    def put_file(self, file_path, key_name, timeout):
-        """
-        Upload a file to an s3 bucket.
-
-        :param file_path: file to upload
-        :param key_name: S3 key to create
-        :param timeout: S3 expiration time in seconds
-        """
-        default_expiration = 30  # minutes
-        expires = datetime.datetime.now() + datetime.timedelta(minutes=default_expiration)
-        if timeout:
-            expires += datetime.timedelta(seconds=timeout)
+def _generate_unique_job_key(job_name):
+    """
+    Generate an unique job key to use as identifier.
 
-        s3_client = self.boto3_factory.get_client("s3")
-        s3_client.upload_file(file_path, self.s3_bucket, key_name, ExtraArgs={"Expires": expires})
+    :param job_name: job name
+    :return: "job-<job_name>-<timestamp>"
+    """
+    return "job-{0}-{1}".format(job_name, int(time.time() * 1000))
 
 
-def _upload_and_get_command(boto3_factory, args, job_name, config, log):
+def _upload_and_get_command(boto3_factory, args, job_folder, job_key, config, log):
     """
     Get command by parsing args and config.
 
     The function will also perform an s3 upload, if needed.
     :param boto3_factory: initialized Boto3ClientFactory object
     :param args: input arguments
-    :param job_name: job name
+    :param job_folder: job folder
+    :param job_key: internal job_key
     :param config: config object
     :param log: log
     :return: command to submit
     """
     if args.command_file or not sys.stdin.isatty() or args.env:
         # define job script name
-        job_id = "job-{0}-{1}".format(job_name, int(time.time() * 1000))
-        job_folder = _get_job_folder(job_id)
-        job_script = job_id + ".sh"
+        job_script = job_key + ".sh"
         log.info("Using command-file option or stdin. Job script name: %s" % job_script)
 
-        s3_uploader = S3Uploader(boto3_factory, config.s3_bucket)
+        s3_uploader = S3Uploader(boto3_factory, config.s3_bucket, job_folder)
         env_file = None
         if args.env:
-            env_file = job_id + ".env.sh"
+            env_file = job_key + ".env.sh"
             # get environment variables and upload file used to extend the submission environment
             env_blacklist = args.env_blacklist if args.env_blacklist else config.env_blacklist
-            _get_env_and_upload(s3_uploader, args.env, env_blacklist, job_folder, env_file, args.timeout, log)
+            _get_env_and_upload(s3_uploader, args.env, env_blacklist, env_file, args.timeout, log)
 
         # upload job script
         if args.command_file:
             # existing script file
             try:
-                s3_uploader.put_file(args.command, job_folder + job_script, args.timeout)
+                s3_uploader.put_file(args.command, job_script, args.timeout)
```

Comment:
Can we just have the ttl on the job_folder so all the stuff down there will inherit the same ttl ?

Reply:
We are removing the Expires property since doesn't work as expected.